src/behavior/impl/balancer-jar.ts

```diff
@@ -3,14 +3,44 @@ import { BigNumber, Contract, ContractTransaction, ethers, Signer } from "ethers
 import strategyAbi from "../../Contracts/ABIs/strategy.json";
 import jarAbi from "../../Contracts/ABIs/jar.json";
 import { JarHarvestStats, PickleModel } from "../..";
-import { HistoricalYield, JarDefinition } from "../../model/PickleModelJson";
-import { getBalancerPerformance } from "../../protocols/BalancerUtil";
+import { AssetProjectedApr, HistoricalYield, JarDefinition } from "../../model/PickleModelJson";
+import { calculateBalPoolAPRs, getBalancerPerformance, getPoolData, PoolData } from "../../protocols/BalancerUtil";
 import { BalancerClaimsManager } from "../../protocols/BalancerUtil/BalancerClaimsManager";
 import { AbstractJarBehavior } from "../AbstractJarBehavior";
 import { Prices } from "../../protocols/BalancerUtil/types";
 import { ICustomHarvester, PfCoreGasFlags } from "../JarBehaviorResolver";
 
-export abstract class BalancerJar extends AbstractJarBehavior {
+export class BalancerJar extends AbstractJarBehavior {
+  poolData: PoolData | undefined;
+
+  async getDepositTokenPrice(
+    jar: JarDefinition,
+    model: PickleModel,
+  ): Promise<number> {
+    if (!this.poolData) this.poolData = await getPoolData(jar, model);
+    return this.poolData.pricePerToken;
```

Comment:
this could throw an NPE... or maybe some other error. Might wanna check that this.poolData exists, but also throw a try-catch around  await getPoolData()